### PR TITLE
Support fish shell completion

### LINKED ISSUE #3281: Add completion for the fish shell
**Describe the problem/challenge you have**

The `velelro` CLI tools currently provides completions for bash and zsh. It would be very useful to also provide completions for fish.

**Describe the solution you'd like**
`velero completion fish` should generate a fish completion script.

**Anything else you would like to add:**
No

**Environment:**

- Velero version (use `velero version`): 1.5.2
- Kubernetes version (use `kubectl version`): Not relevant
- Kubernetes installer & version: Not relevant
- Cloud provider or hardware configuration: Not relevant
- OS (e.g. from `/etc/os-release`): openSUSE Tumbleweed x86_64

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by jenting**
As fish shell supported at [spf13/cobra v1.0.0 release](https://github.com/spf13/cobra/releases/tag/v1.0.0), bump spf13/cobra package version to current latest version v1.1.1 to support fish shell completion for Velero.

The code change reference to https://github.com/spf13/cobra/blob/v1.1.1/shell_completions.md

fixes #3281

### CODE REVIEW

**Review by zubron — CHANGES_REQUESTED** [review]
This is looking great. Thanks for updating the instructions as well. Just a couple of things that 
need to be changed before it can be merged. It's worth noting that this will fix #3281 (opened after this PR).

File: pkg/cmd/cli/completion/completion.go [hunk 1]
```diff
@@ -21,35 +21,56 @@ import (
...
+
+Zsh:
+
+# If shell completion is not already enabled in your environment you will need
+# to enable it.  You can execute the following once:
+
+$ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+# To load completions for each session, execute once:
+$ velero completion zsh > "${fpath[1]}/_yourprogram"
```
[the remark this answers is not in the document]
Comment (jenting) [reply pkg/cmd/cli/completion/completion.go]: nice catch!

**Review by zubron — CHANGES_REQUESTED** [review]
Thanks for making the changes, @jenting! This looks good to me but there is a local file accidentally included in this. Once that's removed, it'll be ready to merge :)

File: credentials-velero [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[default]
```
[the remark this answers is not in the document]
Comment (jenting) [reply credentials-velero]: oops, revert to more it.

**Review by nrb — APPROVED** [review]

**Review by zubron — APPROVED** [review]
Thanks!

**Pull request merged**